contracts/views.py

```diff
@@ -0,0 +1,208 @@
+from django.contrib import messages
+from django.shortcuts import get_object_or_404, redirect, render
+from django.urls import reverse_lazy
+from django.views import View
+from django.views.generic import DeleteView, DetailView, ListView, UpdateView
+
+from properties.forms import TableUnitFilterForm
+from properties.models import Document, Unit
+from users.models import User
+
+from .forms import TenancyContractForm
+from .models import TenancyContract
+
+
+# view for making contract
+class TenancyContractCreateView(View):
+    template_name = 'contracts/contract_create.html'
+    success_url = reverse_lazy('approved_tenants')
+
+    def get(self, request, tenant_id):
+        tenant = get_object_or_404(User, id=tenant_id)
+        owner = request.user
+        unit_id = request.GET.get('unit_id')
+
+        if unit_id:
+            unit = get_object_or_404(Unit, id=unit_id)
+            document = Document.objects.filter(tenant=tenant, unit=unit, status='approved').first()
+            if document:
+                tenancy_contract = TenancyContract.objects.filter(tenant=tenant, owner=owner, unit=unit).first()
+                if tenancy_contract:
+                    form = TenancyContractForm(instance=tenancy_contract)
+                else:
+                    initial_data = {'tenant': tenant, 'owner': owner, 'unit': unit}
+                    form = TenancyContractForm(initial=initial_data)
+            else:
+                form = TenancyContractForm()
+        else:
+            form = TenancyContractForm()
+
+        return render(request, self.template_name, {'form': form})
+
+    def post(self, request, tenant_id):
+        tenant = get_object_or_404(User, id=tenant_id)
+        owner = request.user
+        form = TenancyContractForm(request.POST)
+        if form.is_valid():
+            existing_contract = TenancyContract.objects.filter(tenant=tenant, unit=form.cleaned_data['unit']).first()
+
+            if existing_contract:
+                existing_contract.delete()
+                messages.warning(request, 'New contract created successfully.')
+
+            contract = form.save(commit=False)
+            contract.tenant = tenant
+            contract.owner = owner
+            contract.save()
+            return redirect(self.success_url)
+
+        return render(request, self.template_name, {'form': form})
+
+
+# for active contracts
+class ActiveContractsView(ListView):
```

Comment:
Help function could improve the readability and scalability of the code for future improvements. 

```
def get_filtered_queryset(queryset, request):
    property_id = request.GET.get('property')
    unit_id = request.GET.get('unit')

    if property_id:
        queryset = queryset.filter(unit__property_id=property_id)
    if unit_id:
        queryset = queryset.filter(unit_id=unit_id)

    return queryset

class ActiveContractsView(ListView):
    template_name = 'contracts/active_contracts.html'
    context_object_name = 'contracts'

    def get_queryset(self):
        owner = self.request.user
        queryset = TenancyContract.objects.filter(owner=owner, active=True)
        return get_filtered_queryset(queryset, self.request)

    def get_context_data(self, **kwargs):
        context = super().get_context_data(**kwargs)
        context['filter_form'] = TableUnitFilterForm(user=self.request.user, data=self.request.GET)
        return context

```